engines/sci/engine/seg_manager.h

```diff
@@ -225,9 +225,11 @@ class SegManager : public Common::Serializable {
 	 * @param[in] size		Number of bytes to allocate for the hunk entry
 	 * @param[in] hunk_type	A descriptive string for the hunk entry, for
 	 *						debugging purposes
+	 * @param[in] gc        Whether to make the hunk eligible for garbage
+	 *                      collection
```

Comment:
IMHO, it would be more preferable to store persistent hunk entries in a separate segment type (e.g. PersistentHunk), instead of changing the behavior of the Garbage Collector. Such entries can then be persisted in saved games, as well (like dynamic text objects).